src/main/java/jchess/ruleengine/DefaultRuleProcessor.java

```diff
@@ -44,6 +44,9 @@ public DefaultRuleProcessor(IAppLogger oLogger) {
 	
 	/**
 	 * This method tries to find all the possible candidate moves that the selected Piece can make.
+	 * @param oBoard the chess board
```

Comment:
instead of variable name please mention its type.
E.g. @param IBoardAgent instance for the (active) board

Same is done throughout this file so need to change them all.